src/sentry/static/sentry/app/components/sidebar/help.jsx

```diff
@@ -21,27 +21,6 @@ class SidebarHelp extends React.Component {
     organization: SentryTypes.Organization,
   };
 
-  constructor(props) {
```

Comment:
the reason this was done this way is so that when the user opens the menu there is no lag in the support item showing up compared to other items which are instantaneous. the lag can happen because the subscription object is fetched in an API call in the hook. so instead of loading the item when the menu is opened we load it when the sidebar loads.

Reply:
> fetched in an API call in the hook

It's actually fetched from the subscriptionStore, which will have already likely cached the orgs subscription object via loading from another component (the banner, the settings nav, or anything else).

Reply:
which banner? also settings will only load it if someone has clicked on settings, right? (I understand nbd and this is probably just ok, just wondering how deterministic the caching actually is)

Reply:
![image](https://user-images.githubusercontent.com/1421724/53055178-99e37600-345b-11e9-834d-11485fa23b6b.png)

This one. It will always load the subscription to decide weather or not to show.

> settings will only load it if someone has clicked on settings, right

Yes.

Reply:
ok that only shows on trial btw, which is a very small % of all users

Reply:
It has to hydrate the SubscriptionStore to determine if the user is on a trail though to decide weather or not to show it :)

Reply:
o i c